src/lib/data/stores/localization.js

```diff
@@ -1,10 +1,11 @@
 import { derived } from 'svelte/store';
 import { userSettings } from './setting';
+import { getLanguages } from '$lib/data/language';
 import config from '../config';
 
 /** localization */
-export const languageDefault = config.translationMappings.defaultLang;
-export const languages = Object.keys(config.interfaceLanguages.writingSystems);
+export const languageDefault = userSettings['interface-language']?.defaultValue;
```

Comment:
This won't work if the UserSetting for interface-language is not made available to the end user.